Sources/Token.swift

```diff
@@ -10,27 +10,55 @@
 
 import Foundation
 
-@objcMembers @objc(PAYToken) public final class Token: NSObject {
+@objcMembers @objc(PAYToken) public final class Token: NSObject, Decodable {
     public let identifer: String
     public let livemode: Bool
     public let used: Bool
     public let card: Card
     public let createdAt: Date
-    public let rawValue: Any
- 
-    init (_ e: Extractor) {
-        identifer = try! e <| "id"
-        livemode = try! e <| "livemode"
-        used = try! e <| "used"
-        card = try! e <| "card"
-        createdAt = try! DateTransformer.apply(e <| "created")
-        rawValue = e.rawValue
+    public var rawValue: [String: Any]?
+    
+    // MARK: - Decodable
+    
+    private enum CodingKeys: String, CodingKey {
+        case id = "id"
+        case livemode
+        case used
+        case card
+        case createdAt = "created"
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        identifer = try container.decode(String.self, forKey: .id)
+        livemode = try container.decode(Bool.self, forKey: .livemode)
+        used = try container.decode(Bool.self, forKey: .used)
+        card = try container.decode(Card.self, forKey: .card)
+        createdAt = try container.decode(Date.self, forKey: .createdAt)
     }
 }
 
-extension Token: Decodable {
-    public static func decode(_ e: Extractor) throws -> Self {
-        return try castOrFail(Token(e))
+extension Token {
+    
+    /**
+     * Provide a factory function to decode json by JSONDecoder
+     * and also desereialize all fields as a arbitrary dictionary by JSONSerialization.
+     */
+    static func decodeJson(with data: Data, using decoder: JSONDecoder) throws -> Token {
+        let token = try decoder.decode(self, from: data)
+        // assign rawValue by JSONSerialization
+        typealias RawValue = [String: Any]
```

Comment:
We usually not declare typealias inside any classes, structs or methods. Put it to the top level will be better.

Reply:
Good catch :sob: I've fixed it!